test/integration2/migrate/migration-integration.spec.js

```diff
@@ -527,7 +528,131 @@ describe('Migrations', function () {
           });
         });
 
+        describe('knex.migrate.latest - lifecycle hooks', function () {
```

Comment:
Could you please add tests that:
1. if one of the hooks fails the migration is rolled back
2. If one of the hook fails the next hooks/migrations are not executed:
    1. if `beforeAll` failed, the `beforeEach`, the migration itself, `afterEach` and `afterAll` are not called
    2. if `beforeEach` failed, the migration itself, `afterEach` and `afterAll` are not called
    3. if the migration itself failed, the `afterEach` and `afterAll` are not called
    4. if the `afterEach` failed, `afterAll` is not called
3. test with all the hooks and check execution order - `beforeAll` -> `beforeEach` -> `afterEach` -> `afterAll`
4. no hooks are being called if no pending migrations

Reply:
Great suggestions, thank you!